### PR TITLE
Bump ruby to 3.2.2 & bundler to 2.4.22

### PR DESCRIPTION
**Pull request by dombesz**
https://community.openproject.org/wp/51395
As part of the rails upgrade, it is recommended to get ruby to the latest version.

The failing spec is related to the new old chrome headless version reporting itself with a different name. The issue is being addressed in a separate PR #14426 .

### CODE REVIEW

**Comment by aaron-contreras** [comment]
:) https://community.openproject.org/projects/openproject/work_packages/51395/activity

File: Gemfile [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 source 'https://rubygems.org'
 
-ruby '~> 3.2.1'
+ruby file: '.ruby-version'
```
Comment (aaron-contreras) [inline Gemfile:31]: I didn't know about this!

**Comment by klaustopher** [comment]
Just FYI, on the Dec 25th 3.3 will be released. Maybe we can already jump on the RC1 and then move to the released version net year? It brings some more performance increases. See https://www.ruby-lang.org/en/news/2023/12/11/ruby-3-3-0-rc1-released/

File: Gemfile (hunk 1 above)
Comment (aaron-contreras) [reply Gemfile:31]: Looks like it's not being read on the CI based on the failed build. How about: https://andycroll.com/ruby/read-ruby-version-in-your-gemfile/

```ruby
ruby File.read(".ruby-version").strip
```

Comment (dombesz) [reply Gemfile:31]: The syntax is part of the new bundler release. I assume there are some references to the old bundler, hence the error.

**Comment by oliverguenther** [comment]
> Just FYI, on the Dec 25th 3.3 will be released. Maybe we can already jump on the RC1 and then move to the released version net year? It brings some more performance increases. See https://www.ruby-lang.org/en/news/2023/12/11/ruby-3-3-0-rc1-released/

I'd be in favor of upgrading ruby pessimistively, we've had some problems with .0 versions in the past, so I'd wait until it settled down. If the bump is easy, there's not much effort involved in upgrading twice in 13.2

File: script/github_pr_errors [hunk 2]
```diff
@@ -272,7 +272,7 @@ end
...
+# Process 28: TEST_ENV_NUMBER=28 RUBYOPT=-I/usr/local/bundle/bundler/gems/turbo_tests-3148ae6c3482/lib -r/usr/local/bundle/gems/bundler-2.4.22/lib/bundler/setup -W0 RSPEC_SILENCE_FILTER_ANNOUNCEMENTS=1 /usr/local/bundle/gems/bundler-2.4.22/exe/bundle exec rspec --seed 52674 --format TurboTests::JsonRowsFormatter --out tmp/test-pipes/subprocess-28 --format ParallelTests::RSpec::RuntimeLogger --out spec/support/turbo_runtime_features.log spec/features/api_docs/index_spec.rb spec/features/custom_fields/reorder_options_spec.rb spec/features/projects/projects_portfolio_spec.rb spec/features/projects/template_spec.rb spec/features/versions/edit_spec.rb spec/features/work_packages/details/markdown/description_editor_spec.rb spec/features/work_packages/table/hierarchy/hierarchy_parent_below_spec.rb spec/features/work_packages/table/inline_create/inline_create_refresh_spec.rb spec/features/work_packages/table/invalid_query_spec.rb spec/features/work_packages/tabs/activity_revisions_spec.rb
```
Comment (aaron-contreras) [inline script/github_pr_errors:275]: 👍

**Review by aaron-contreras — APPROVED** [review]
Awesome!👏 I love the new Bundler syntax for specifying the ruby version in the Gemfile with your rbenv-generated `.ruby-version` file. The failing spec is already being addressed in another PR and is unrelated to the upgrade.

```
                     `. ___
                    __,' __`.                _..----....____
        __...--.'``;.   ,.   ;``--..__     .'    ,-._    _.-'
  _..-''-------'   `'   `'   `'     O ``-''._   (,;') _,'
,'________________                          \`-._`-','
 `._              ```````````------...___   '-.._'-:
    ```--.._      ,.                     ````--...__\-.
            `.--. `-`                       ____    |  |`
              `. `.                       ,'`````.  ;  ;`
                `._`.        __________   `.      \'__/`
                   `-:._____/______/___/____`.     \  `
                               |       `._    `.    \
                               `._________`-.   `.   `.___
                                             SSt  `------'`
```

**Comment by cbliard** [comment]
I'll merge it now, as the failing spec is unrelated to this PR.

**Pull request merged**